src/Security/UserProvider/OidcUserProviderInterface.php

```diff
@@ -4,13 +4,14 @@
 
 use Drenso\OidcBundle\Exception\OidcException;
 use Drenso\OidcBundle\Model\OidcUserData;
+use Lcobucci\JWT\Token\DataSet;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Security\Core\User\UserProviderInterface;
 
 interface OidcUserProviderInterface extends UserProviderInterface
 {
   /** @throws OidcException Can be thrown when the user cannot be created */
-  public function ensureUserExists(string $userIdentifier, OidcUserData $userData);
+  public function ensureUserExists(string $userIdentifier, OidcUserData $userData, ?DataSet $idTokenDataSet = null);
```

Comment:
Hmm, would it make sense to pass the `OidcTokens` instead? Than you can access both tokens, but it does make the processing the responsibility of the user.

Reply:
yeap, users will need to decode themselves. And we already have/need decoded versions in OidcAuthenticator

Reply:
Not really, it is only decoded when a specific option is used. Also, passing the complete token data could also open up an option to use those tokens in whatever additional request you might want to do in the `ensureUserExists` method.

Can you change it to pass `OidcTokens` instead?

Reply:
applied changes